lib/graphql/argument.rb

```diff
@@ -54,9 +72,22 @@ def expose_as
       @expose_as ||= (@as || @name).to_s
     end
 
+    # @param value
+    # @return [Object] The prepared `value` for this argument or `value` itself if no `prepare` function exists.
+    def prepare(value)
+      return value unless @prepare_proc.respond_to?(:call)
+      @prepare_proc.call(value)
```

Comment:
How about removing this logical branch by providing a default, no-op `@prepare_proc`? For example:

```ruby
module DefaultPrepare 
  def self.call(value); value; end 
end 

def initialize 
  # ... 
  @prepare_proc = DefaultPrepare
end 

def prepare(value)
  @prepare_proc.call(value)
end 
```

It requires some setup, but it simplifies the runtime code since we can always assume that a `@prepare_proc` is present.